pyglmnet/pyglmnet.py

```diff
@@ -747,12 +749,383 @@ def score(self, X, y):
+                ('verbose', self.verbose)
+            )
+        )
+
+    def __repr__(self):
+        """Description of the object."""
+        reg_lambda = self.reg_lambda
+        s = '<\nDistribution | %s' % self.distr
+        s += '\nalpha | %0.2f' % self.alpha
+        s += '\nmax_iter | %0.2f' % self.max_iter
+        if len(reg_lambda) > 1:
+            s += ('\nlambda: %0.2f to %0.2f\n>'
+                  % (reg_lambda[0], reg_lambda[-1]))
+        else:
+            s += '\nlambda: %0.2f\n>' % reg_lambda[0]
+        return s
+
+    def copy(self):
+        """Return a copy of the object.
+
+        Parameters
+        ----------
+        none:
+
+        Returns
+        -------
+        self: instance of GLM
+            A copy of the GLM instance.
+        """
+        return deepcopy(self)
+
+    def fit(self, X, y):
+        """The fit function.
+        Parameters
+        ----------
+        X : array
+            The input data of shape (n_samples, n_features)
+
+        y : array
+            The target data
+
+        Returns
+        -------
+        self : instance of GLM
+            The fitted model.
+        """
+
+        logger.info('Looping through the regularization path')
+        glms, scores = list(), list()
+        self.ynull_ = np.mean(y)
+
+        for l, rl in enumerate(self.reg_lambda):
+            logger.info('Lambda: %6.4f' % rl)
+
+            glm = GLM(distr=self.distr,
+                      alpha=self.alpha,
+                      Tau=self.Tau,
+                      reg_lambda=rl,
+                      solver=self.solver,
+                      learning_rate=self.learning_rate,
+                      max_iter=self.max_iter,
+                      tol=self.tol,
+                      eta=self.eta,
+                      score_metric=self.score_metric,
+                      random_state=self.random_state,
+                      verbose=self.verbose)
+
+            if l == 0:
+                glm.beta0_, glm.beta = self.beta0_, self.beta_
+            else:
+                glm.beta0_, glm.beta_ = glms[-1].beta0_, glms[-1].beta_
+
+            glm.fit(X, y)
+            glms.append(glm)
+            scores.append(glm.score(X, y))
+
+        # Update the estimated variables
+        if self.score_metric == 'deviance':
+            opt = np.array(scores).argmin()
+        elif self.score_metric in ['pseudo_R2', 'accuracy']:
+            opt = np.array(scores).argmax()
+        self.beta0_, self.beta_ = glms[opt].beta0_, glms[opt].beta_
+        self.reg_lambda_opt = self.reg_lambda[opt]
+        return self
+
+    def predict(self, X):
+        """Predict targets.
+
+        Parameters
+        ----------
+        X : array
+            Input data for prediction, of shape (n_samples, n_features)
+
+        Returns
+        -------
+        yhat : array
+            The predicted targets of shape (n_samples,)
+        """
+        glm = GLM(distr=self.distr,
```

Comment:
i don't create `self.glm_` in `fit()`. it' a tradeoff: done this way, we don't need to store anything more as class atributes. if we store, we don't need to instantiate.